data/campaigns/Descent_Into_Darkness/scenarios/06_Return_to_Parthyn.cfg

```diff
@@ -259,53 +279,242 @@
+
+                if saved_orc_leader ~= nil then
+                    saved_orc_leader.side = 2
+                    saved_orc_leader.facing = "se"
+
+                    local loc = wesnoth.get_starting_location(2)
+                    wesnoth.put_unit(saved_orc_leader, loc[1], loc[2])
+
+                    wml.variables.orc_leader_store = nil
+                else
+                    local loc = wesnoth.get_starting_location(2)
+                    wesnoth.put_unit(fallback_orc_leader, loc[1], loc[2])
+                end
+            >>
+        [/lua]
+
+        [terrain]
+            x,y=30,4
+            terrain=Ke
+        [/terrain]
+        [terrain]
+            x=30,30,29,29
+            y=3,5,4,5
+            terrain=Ce
+        [/terrain]
+        [redraw][/redraw]
+
+        [if]
+            {DID_RTP_HAS_DROGAN}
+            [then]
+                [message]
+                    side=2
+                    canrecruit=yes
+                    message= _ "Grunts! Prepare the camp! We attack the city tonight— wait. You! Necromancer, you followed us all the way here?!"
+                [/message]
+
+                [message]
+                    speaker=Malin Keshar
+                    message= _ "Did you really think you could outrun the dead, who need no rest nor sleep? Prepare to meet your makers, orc. Drogan, I will show you what my new powers are capable of. You will see how foolish you are to oppose me."
+                [/message]
+            [/then]
+            [else]
+                [message]
+                    side=2
+                    canrecruit=yes
+                    message= _ "Grunts! Prepare the camp! We attack the city tonight— wait. You! Necromancer, you followed us all the way here?!"
+                [/message]
+
+                [message]
+                    speaker=Malin Keshar
+                    message= _ "Did you really think you could outrun the dead, who need no rest nor sleep? Prepare to meet your makers, orc."
+                [/message]
+            [/else]
+        [/if]
+
+        [show_objectives][/show_objectives]
+    [/event]
+
+
     # In-scenario Events
+
+    [event]
+        name=recruit,recall
+        [filter]
+            side=1
+            type=Dark Adept DiD,Dark Sorcerer DiD,Necromancer
+        [/filter]
+
+        [if]
+            {DID_RTP_HAS_DROGAN}
+            [then]
+                [message]
+                    speaker=Drogan
+                    message= _ "So, Malin, you have even begun to corrupt our people with your sorcerous ways. Even I did not expect you to stoop low enough to kidnap the villagers and make them into your pawns!"
+                [/message]
+
+                [message]
+                    speaker=unit
+                    message= _ "I joined Malin of my own free will! Drogan, you are a fool who would let the orcs slay us all simply because you do not understand the power that necromancy holds."
```

Comment:
Thanks for adding this, it sounds good.

It should probably have a female_message too.

Reply:
Added, did I do it properly?

Reply:
The (AFAIK undocumented) coding convention seems to be to use `message=` for the male message.

Reply:
Sorry to keep adding comments, but please consider whether it should have an [allow_undo][/allow_undo], like the hint in #3422 (which would allow undoing a recall, but not a recruit).

I'm not sure if it should or shouldn't have this - strategy-wise, it seems better to allow it; but story-wise it's another irrevocable small step on Malin's descent.

Reply:
I don't think I should allow undoing because I feel like it would look really strange if the text fires but then you just undo recalling your unit.